include/zephyr/bluetooth/gatt.h

```diff
@@ -1343,6 +1355,9 @@ struct bt_gatt_discover_params {
 	/** Only for stack-internal use, used for automatic discovery. */
 	struct bt_gatt_subscribe_params *sub_params;
 #endif /* defined(CONFIG_BT_GATT_AUTO_DISCOVER_CCC) */
+#ifdef CONFIG_BT_EATT
+	enum bt_att_chan_option chan_option;
+#endif /* CONFIG_BT_EATT */
```

Comment:
I am not sure the GATT discovery should be even possible for the user to be done over EATT. It's because the fact the EATT is supported by remote or not is based on the knowledge whether remote has  GATT `SERVER SUPPORTED FEATURES` characteristic value and `EATT Supported` bit is set.

Thus my assumption is that GATT Discovery shall be done over unenhenced bearer only.

BTW, does the implementation reads the `SERVER SUPPORTED FEATURES` before trying to connect EATT channels?

Reply:
It is not required for LE to read the server features before connecting EATT channels and we don't do that currently.

From 6.2 LE GAP INTEROPERABILITY REQUIREMENTS:
> Note: Unlike BR/EDR, it is not necessary to check the Server Supported
Features characteristic before attempting to establish an Enhanced ATT
bearer.

I am not sure how useful discovery over EATT is, but not sure we should disallow it either.

Reply:
@MariuszSkamra ping

Reply:
OK :)